lib/go-qmstr/database/fileNode.go

```diff
@@ -266,3 +266,25 @@ func (db *DataBase) GetFileNodeHashByPath(path string) (string, error) {
 	}
 	return ret["hasNode"][0].FileData.GetHash(), nil
 }
+
+// GetFileNodeLeaves returns the filenodes that are not derived from other filenodes
+func (db *DataBase) GetFileNodeLeaves() ([]*service.FileNode, error) {
+	var ret map[string][]*service.FileNode
+
+	q := `query fileNodeLeaves(){
+		  filenodeleaves(func: has(fileNodeType)) @filter(eq(count(derivedFrom),0) {
+			uid
+			path
```

Comment:
Include also fileData in the query cause analyzers are going to connect the info nodes to the file data nodes.